### PR TITLE
Return Promise when user not pass the callback function

### PR DESCRIPTION
**Pull request by wingyplus**
Hi 

I propose an idea about promise api. When user doesn't pass the callback function, it's return promise automatically.

If you have alternative idea, please suggest.

Thanks

### CODE REVIEW

File: test/main.js [hunk 1]
```diff
@@ -59,4 +59,16 @@ describe("Main test", function() {
...
     });
+
+    it("return promise when does not pass the done function", function(done) {
+        xejs(__dirname + "/file1.md", config.options, config.args).then(function(res) {
+            done();
+        });
+    });
+
+    it("return rejected promise when has an error", function(done) {
+        xejs(__dirname + "/notfound.md", config.options, config.args).then(function() {}, function(err) {
```
Comment (angrykoala) [inline test/main.js]: The argument err should have value? If so, please assert in case it doesn't

File: test/main.js [hunk 2]
```diff
@@ -59,4 +59,16 @@ describe("Main test", function() {
             });
         });
     });
+
+    it("return promise when does not pass the done function", function(done) {
+        xejs(__dirname + "/file1.md", config.options, config.args).then(function(res) {
```
Comment (angrykoala) [inline test/main.js]: assert res exists please

File: index.js [hunk 3]
```diff
@@ -47,4 +47,13 @@ module.exports = function(file, renderingOptions, args, done) {
         err=e;
     }
     if(done) return done(err,res);
+    else {
+        return new Promise(function(resolve, reject) {
+            if (err) {
+              reject(err);
+              return;
```
Comment (angrykoala) [inline index.js]: Could using `if ... else` make the code a bit cleaner by avoiding the `return`?

**Comment by wingyplus** [comment]
@angrykoala I fix 3 issues that you reviewed.

**Pull request merged**